frontend/src/pages/warehouse/Warehouses.js

```diff
@@ -0,0 +1,199 @@
+        warehouse: []
+    });
+    const [errorMessage, setErrors] = useState('');
+
+    const handleCheckedChange = (warehouseId) => {
+        const index = checkBoxes.indexOf(warehouseId);
+        if (index > -1) {
+            checkBoxes.splice(index, 1);
+        } else {
+            checkBoxes.push(warehouseId);
+        }
+    };
+
+    const handleCountPerPage = (e) => {
+        e.preventDefault();
+        setPage(preState => ({
+            ...preState,
+            countPerPage: e.target.value
+        }));
+        getWarehouses(`/customers/${currentCustomerId}/warehouses?size=${e.target.value}`);
+    };
+
+    const changePage = (e) => {
+        e.preventDefault();
+        let currentPage = e.target.innerHTML - 1;
+        getWarehouses(`/customers/${currentCustomerId}/warehouses?page=${currentPage}&size=${page.countPerPage}`);
+    };
+
+    useEffect(() => {
+        getWarehouses(`/customers/${currentCustomerId}/warehouses?size=${page.countPerPage}`);
+    }, []);
+
+    function getWarehouses(url) {
+        fetch(url)
+            .then(response => response.json())
+            .then(commits => {
+                setWarehouses(commits.content);
+                setPage({
+                        countPerPage: commits.size,
+                        countPages: commits.totalPages
+                    }
+                );
+            });
+    }
+
+    const closeModalEdit = (e, warehouseDto) => {
+
+        setEditWarehouse(
+            preState => ({
+                ...preState,
+                editShow: false
+            }));
+        if (warehouseDto) {
+            getWarehouses(`/customers/${currentCustomerId}/warehouses?size=${page.countPerPage}`);
+        }
+    };
+
+    const closeModalAdd = (e, warehouseDto) => {
+        setLgShow(e);
+        if (warehouseDto) {
+            getWarehouses(`/customers/${currentCustomerId}/warehouses?size=${page.countPerPage}`);
+        }
+    };
+
+    function deleteWarehouse() {
+        fetch(`/customers/${currentCustomerId}/warehouses/delete-list`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(checkBoxes)
+        })
+            .then(response => {
+                if (response.status !== 204) {
+                    setErrors({
+                        errorMessage: "Warehouse can not be deleted, because it is already used in application"
+                    })
+                } else {
+                    fetch(`/customers/${currentCustomerId}/users/delete-list`, {
+                        method: 'POST',
+                        headers: {
+                            'Content-Type': 'application/json'
+                        },
+                        body: JSON.stringify(checkBoxes)
+                    })
+                    setCheckBox([]);
+                    getWarehouses(`/customers/${currentCustomerId}/warehouses?size=${page.countPerPage}`);
```

Comment:
It seems like 2 operations above are executed regardless of the result of delete-list operation. Should there be dependency between them?

Reply:
no, they aren't related

Reply:
so why if delete is not successful we uncheck on 114 line?